contrib/storage-hive/core/src/test/java/org/apache/drill/exec/store/hive/schema/TestSchemaConversion.java

```diff
@@ -0,0 +1,184 @@
+            .resumeSchema()
+        .buildSchema()
+        .metadata(0);
+    verifyConversion("struct<t1:boolean,t2:int>", expectedSchema);
+
+    expectedSchema = new SchemaBuilder()
+        .addMap("a")
+            .addNullable("t1", TypeProtos.MinorType.BIT)
+            .addMap("t2")
+                .addNullable("t3", TypeProtos.MinorType.VARDECIMAL, 38, 8)
+                .resumeMap()
+            .resumeSchema()
+        .buildSchema()
+        .metadata(0);
+    verifyConversion("struct<t1:boolean,t2:struct<t3:decimal(38,8)>>", expectedSchema);
+  }
+
+  @Test
+  public void testRepeatedSchema() {
+    verifyConversion("array<boolean>", Types.repeated(TypeProtos.MinorType.BIT));
+
+    ColumnMetadata expectedSchema = new SchemaBuilder()
+        .addRepeatedList("a")
+            .addArray(TypeProtos.MinorType.BIT)
+            .resumeSchema()
+        .buildSchema()
+        .metadata(0);
+    verifyConversion("array<array<boolean>>", expectedSchema);
+
+    expectedSchema = new SchemaBuilder()
+        .addRepeatedList("a")
+            .addDimension()
+                .addArray(TypeProtos.MinorType.BIT)
+                .resumeList()
+            .resumeSchema()
+        .buildSchema()
+        .metadata(0);
+    verifyConversion("array<array<array<boolean>>>", expectedSchema);
+  }
+
+  @Test
+  public void testRepeatedStructSchema() {
+    ColumnMetadata expectedSchema = new SchemaBuilder()
+        .addMapArray("a")
+            .addNullable("t1", TypeProtos.MinorType.VARCHAR, 999)
+            .addNullable("t2", TypeProtos.MinorType.INT)
+            .resumeSchema()
+        .buildSchema()
+        .metadata(0);
+    verifyConversion("array<struct<t1:varchar(999),t2:int>>", expectedSchema);
+
+    expectedSchema = new SchemaBuilder()
+        .addRepeatedList("a")
+            .addMapArray()
+                .addNullable("t1", TypeProtos.MinorType.VARCHAR, 999)
+                .addNullable("t2", TypeProtos.MinorType.INT)
+                .resumeList()
+            .resumeSchema()
+        .buildSchema()
+        .metadata(0);
+    verifyConversion("array<array<struct<t1:varchar(999),t2:int>>>", expectedSchema);
+
+    expectedSchema = new SchemaBuilder()
+        .addRepeatedList("a")
+            .addDimension()
+                .addMapArray()
+                    .addRepeatedList("t1")
+                        .addArray(TypeProtos.MinorType.VARCHAR, 999)
+                        .resumeMap()
+                    .addArray("t2", TypeProtos.MinorType.VARDECIMAL, 28, 14)
+                    .resumeList()
+                .resumeList()
+            .resumeSchema()
+        .buildSchema()
+        .metadata(0);
+    verifyConversion("array<array<array<struct<t1:array<array<varchar(999)>>,t2:array<decimal(28,14)>>>>>", expectedSchema);
+  }
+
+  @Test
+  public void testUnionSchema() {
+    thrown.expect(UnsupportedOperationException.class);
+    thrown.expectMessage(containsString("Unsupported data type"));
```

Comment:
Nit: checking exception message is brittle. I would say checking that we receive `UnsupportedOperationException` is enough.

Reply:
Thanks for pointing this, removed redundant checks.